### PR TITLE
916 sample loader single source

### PR DESCRIPTION
**Pull request by LukeLoze1**
# Motivation and Context
Imports and uses census-rm-sample-loader

# What has changed
no longer has it's own duplicated sample loader and RabbitContext code

# How to test?
This branch currently imports the correct version of census-rm-sample-loader
Run pipenv install --dev to use it then....
make test  (have a standard docker dev up and running first)

# Links
https://trello.com/c/gbDQdP2E/916-update-census-rm-sample-loader-5
https://github.com/ONSdigital/census-rm-sample-loader/tree/916-source-of-truth

# Screenshots (if appropriate):

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -3,4 +3,5 @@ FROM python:3.6-slim
 WORKDIR /app
 COPY . /app
 RUN apt-get update -y && apt-get install -y python-pip && apt-get install -y curl
+RUN apt-get install -y git
```
Comment (jcox-dev) [inline Dockerfile]: This should be tagged onto the end of the line above to save on layers.

**Review by rjweeks — APPROVED** [review]

**Review by jcox-dev — CHANGES_REQUESTED** [review]

File: acceptance_tests/features/environment.py [hunk 2]
```diff
@@ -2,7 +2,7 @@
 import json
 import uuid
 from datetime import datetime
-from acceptance_tests.utilities.rabbit_context import RabbitContext
+from rabbit_context import RabbitContext
```
Comment (jcox-dev) [inline acceptance_tests/features/environment.py]:
```suggestion

from rabbit_context import RabbitContext

```

File: acceptance_tests/features/steps/receipt.py [hunk 3]
```diff
@@ -8,7 +8,7 @@
 from google.cloud import pubsub_v1
 
 from acceptance_tests.utilities.date_utilities import convert_datetime_to_str
-from acceptance_tests.utilities.rabbit_context import RabbitContext
+from rabbit_context import RabbitContext
```
Comment (jcox-dev) [inline acceptance_tests/features/steps/receipt.py]: Should sit in same block as the google packages, etc.

**Review by jcox-dev — DISMISSED** [review]

**Review by ryangrundy7 — COMMENTED** [review]
Code looks good, will test it once the merge conflicts have been addressed

File: Pipfile [hunk 4]
```diff
@@ -14,6 +14,8 @@ retrying = "*"
 pika = "*"
 paramiko = "*"
 "flake8" = "*"
+census-rm-sample-loader = {editable = true,git = "https://github.com/ONSdigital/census-rm-sample-loader",ref = "916-source-of-truth"}
```
Comment (ryangrundy7) [inline Pipfile]: Got to remember to remove the `ref` once sample loader has been merged in

Comment (AdamHawtin) [reply Pipfile]: And specify a release tag

Comment (LukeLoze1) [reply Pipfile]: yeap I'll do this when sample-loader is released

File: acceptance_tests/features/environment.py [hunk 5]
```diff
@@ -2,7 +2,8 @@
...
```
Comment (AdamHawtin) [inline acceptance_tests/features/environment.py]: I'm not sure about this, using the rabbit context through the sample loader feels like bad practice. If we're going to reuse the rabbit wrapper code I think it should be pulled out into a library and made properly reusable, then passed into whatever code needs to use it.

**Review by AdamHawtin — DISMISSED** [review]

File: Dockerfile [hunk 6]
```diff
@@ -2,5 +2,5 @@ FROM python:3.6-slim
 
 WORKDIR /app
 COPY . /app
-RUN apt-get update -y && apt-get install -y python-pip && apt-get install -y curl
+RUN apt-get update -y && apt-get install -y python-pip && apt-get install -y curl && apt-get install -y git
```
Comment (AdamHawtin) [inline Dockerfile]:
```suggestion
RUN apt-get update -y && apt-get install -y python-pip curl git
```

File: acceptance_tests/features/environment.py (hunk 5 above)
Comment (LukeLoze1) [reply acceptance_tests/features/environment.py]: I'm not 100% convinced about any of it.  I've put back the local one for ATs for now

File: acceptance_tests/features/environment.py [hunk 7]
```diff
@@ -18,7 +19,7 @@ def before_all(_):
 
 
 def after_all(context):
-    with RabbitContext() as rabbit:
+    with RabbitContext(port=Config.RABBITMQ_PORT) as rabbit:
```
Comment (AdamHawtin) [inline acceptance_tests/features/environment.py]: `Config.RABBITMQ_PORT` is the default value.

File: acceptance_tests/features/environment.py [hunk 8]
```diff
@@ -33,7 +34,7 @@ def before_scenario(context, _):
 
 
 def _purge_queues():
-    with RabbitContext() as rabbit:
+    with RabbitContext(port=Config.RABBITMQ_PORT) as rabbit:
```
Comment (AdamHawtin) [inline acceptance_tests/features/environment.py]: `Config.RABBITMQ_PORT` is the default value.

File: acceptance_tests/features/steps/receipt.py [hunk 9]
```diff
@@ -40,7 +40,7 @@ def create_receipt_received_message_from_eq(context):
     # First case in list will be receipted
     context.receipted_case_id = context.case_created_events[0]['payload']['collectionCase']['id']
 
-    with RabbitContext(queue_name=Config.RABBITMQ_INBOUND_EQ_QUEUE) as rabbit:
+    with RabbitContext(queue_name=Config.RABBITMQ_INBOUND_EQ_QUEUE, port=Config.RABBITMQ_PORT) as rabbit:
```
Comment (AdamHawtin) [inline acceptance_tests/features/steps/receipt.py]: `Config.RABBITMQ_PORT` is the default value.

File: acceptance_tests/features/steps/unaddressed_steps.py [hunk 10]
```diff
@@ -17,7 +17,7 @@
 @when('an unaddressed message of questionnaire type {questionnaire_type} is sent')
 def send_unaddressed_message(context, questionnaire_type):
     context.expected_questionnaire_type = questionnaire_type
-    with RabbitContext(queue_name=Config.RABBITMQ_UNADDRESSED_REQUEST_QUEUE) as rabbit:
+    with RabbitContext(queue_name=Config.RABBITMQ_UNADDRESSED_REQUEST_QUEUE, port=Config.RABBITMQ_PORT) as rabbit:
```
Comment (AdamHawtin) [inline acceptance_tests/features/steps/unaddressed_steps.py]: `Config.RABBITMQ_PORT` is the default value.

File: acceptance_tests/utilities/rabbit_helper.py [hunk 11]
```diff
@@ -3,16 +3,15 @@
...
-    RabbitContext
-)
+from config import Config
 
 logger = wrap_logger(logging.getLogger(__name__))
 
 
 def start_listening_to_rabbit_queue(queue, on_message_callback, timeout=30):
-    rabbit = RabbitContext(queue_name=queue)
+    rabbit = RabbitContext(queue_name=queue, port=Config.RABBITMQ_PORT)
```
Comment (AdamHawtin) [inline acceptance_tests/utilities/rabbit_helper.py]: `Config.RABBITMQ_PORT` is the default value.

File: acceptance_tests/utilities/rabbit_helper.py [hunk 12]
```diff
@@ -45,7 +44,7 @@ def store_all_msgs_in_context(ch, method, _properties, body, context, expected_m
 
 
 def add_test_queue(binding_key, exchange_name, queue_name, exchange_type='topic'):
-    with RabbitContext() as rabbit:
+    with RabbitContext(port=Config.RABBITMQ_PORT) as rabbit:
```
Comment (AdamHawtin) [inline acceptance_tests/utilities/rabbit_helper.py]: `Config.RABBITMQ_PORT` is the default value.

File: acceptance_tests/features/steps/unaddressed_steps.py (hunk 10 above)
Comment (LukeLoze1) [reply acceptance_tests/features/steps/unaddressed_steps.py]: good spot

**Review by ryangrundy7 — APPROVED** [review]
All tests pass locally and on GCP 👍

**Review by liamtoozer — APPROVED** [review]
LGTM 👍

**Pull request merged**